### DISCUSSION TITLE
Can I use Xtion Live Pro for HSRC ?　(Xtion mulfunction)

### DISCUSSION THREAD

**Discussion by s-kajita** [HumanSupportRobot/QandA#87] [Q&A] [answered]
The HSRC in my lab does not boot normally. It displays an error 

/Sensors/Camera/HeadRGBDSensor/...
boot time out  

Since the red LED of Xtion sensor doesn't turn on,  it might be broken.
For replacement, can I use "Xtion Live Pro" ? 

(That specific type is available on the internet shop)

**Comment by yoshimi-iyoda** [comment, marked as answer]
Xtion may be out of order, but would you be willing to let us check it out for us to investigate?
Could you run the lsusb command as follows and see if you get "1d27:0601 ASUS"?

```
$ lsusb
:
:
Bus 001 Device 009: ID 1d27:0601 ASUS
:
:
```

If you see the Xtion device (1d27:0601), please follow the steps below to see if the log is output.

1. Execute the journal command

    `$ journalctl -f | grep "Device is in safe mode"`

2. Release the stop button.

**Reply by s-kajita** [reply]
Here are the results.

 $ lsusb
 :
 Bus 001 Device 009: ID 1d27:0600 ASUS 
 :
 
 $ journalctl -f | grep "Device is in safe mode"
 
 (Releasing the stop button)
11月 17 17:35:42 hsrc docker.hsrb.robot.service[3271]:         Could not open "1d27/0600@1/9": Device is in safe mode. Cannot start any stream!
                                             Could not open "1d27/0600@1/9": Device is in safe mode. Cannot start any stream!
                                             Could not open "1d27/0600@1/9": Device is in safe mode. Cannot start any stream!
11月 17 17:35:45 hsrc docker.hsrb.robot.service[3271]:         Could not open "1d27/0600@1/9": Device is in safe mode. Cannot start any stream!
11月 17 17:35:48 hsrc docker.hsrb.robot.service[3271]:         Could not open "1d27/0600@1/9": Device is in safe mode. Cannot start any stream!
                                             Could not open "1d27/0600@1/9": Device is in safe mode. Cannot start any stream!
11月 17 17:35:51 hsrc docker.hsrb.robot.service[3271]:         Could not open "1d27/0600@1/9": Device is in safe mode. Cannot start any stream!
                                             Could not open "1d27/0600@1/9": Device is in safe mode. Cannot start any stream!
11月 17 17:35:54 hsrc docker.hsrb.robot.service[3271]:         Could not open "1d27/0600@1/9": Device is in safe mode. Cannot start any stream!
                                             Could not open "1d27/0600@1/9": Device is in safe mode. Cannot start any stream!
11月 17 17:35:57 hsrc docker.hsrb.robot.service[3271]:         Could not open "1d27/0600@1/9": Device is in safe mode. Cannot start any stream!
                                             Could not open "1d27/0600@1/9": Device is in safe mode. Cannot start any stream!
                                             Could not open "1d27/0600@1/9": Device is in safe mode. Cannot start any stream!

**Reply by yoshimi-iyoda** [reply]
Thank you for running the command.
We have had similar events at other locations in the past, so I will check the recovery procedure we implemented at that time.

**Reply by s-kajita** [reply]
That sounds great!  I will be waiting for your information.

**Reply by yoshimi-iyoda** [reply]
I have been informed that the Xtion firmware update has restored the system.

Please download the firmware from the following URL and try updating it.
https://www.asus.com/lk/supportonly/xtion%20pro/helpdesk_bios/



However, there are some cases where the firmware update does not fix the problem, as shown below.
https://stackoverflow.com/questions/8702320/openni-open-failed-device-is-in-safe-mode-cannot-start-any-stream

**Reply by s-kajita** [reply]
Thank you very much!   We will try it.

**Reply by s-kajita** [reply]
We believe we have updated the firmware successfully (as shown in the screenshot), but the problem was not fixed...

(To run the firmware updater, we need old Windows 8 computer, and we tried the updating several times. )
 
![IMG_7861](https://github.com/HumanSupportRobot/QandA/assets/10680935/7d6924a8-75ae-4c7b-b68f-cfa54d987c7c)

Uploading IMG_7861.MOV…

**Reply by yoshimi-iyoda** [reply]
I apologize for my inability to help you.
Please wait a moment as I will look into the possibility of replacing the Xtion.
